observer/observer.go

```diff
@@ -152,14 +158,56 @@ func (o *Observer) Watch(files []string) error {
 	return nil
 }
 
-// handleEvent handle an event.
-func (o *Observer) handleEvent(event interface{}) {
-	// Run all listeners for this event.
+// SetBufferDuration set the buffer damping duration
+func (o *Observer) SetBufferDuration(d time.Duration) {
+	// Create the buffer mutex, if missing.
+	if o.bufferMutex == nil {
+		o.bufferMutex = &sync.Mutex{}
+	}
+
+	// Set the buffer duration.
+	o.bufferDuration = d
+}
+
+// sendEvent send one or more events to the observer listeners.
+func (o *Observer) sendEvent(event interface{}) {
 	for _, listener := range o.listeners {
 		go listener(event)
 	}
 }
 
+// handleEvent handle an event.
+func (o *Observer) handleEvent(event interface{}) {
+	// If we do not buffer events, just send this event now.
+	if o.bufferDuration == 0 {
+		o.sendEvent(event)
+		return
+	}
+
+	// Lock this function.
+	o.bufferMutex.Lock()
+	defer o.bufferMutex.Unlock()
+
+	// Add new event to the event buffer
+	o.bufferEvents = append(o.bufferEvents, event)
+
+	// If this is the first event, set a timeout function.
+	if o.bufferTimer == nil {
+		o.bufferTimer = time.AfterFunc(o.bufferDuration, func() {
+			// Lock this function.
+			o.bufferMutex.Lock()
```

Comment:
I'm missing here something. You already have a lock on line 188.

Based on the documentation:
```
Lock locks m. If the lock is already in use, the calling goroutine blocks until the mutex is available. 
```
So why lock it above?